src/workerd/api/actor-state.c++

```diff
@@ -919,6 +919,36 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(kj::Maybe<int> timeoutMs) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+
+  KJ_IF_SOME(timeout, timeoutMs) {
+    if (timeout > 0) {
+      if (a.getHibernationManager() == kj::none) {
+        a.setHibernationManager(kj::refcounted<HibernationManagerImpl>(
+                a.getLoopback(), KJ_REQUIRE_NONNULL(a.getHibernationEventType())));
+      }
+      KJ_REQUIRE_NONNULL(a.getHibernationManager()).setEventTimeout(timeout);
```

Comment:
Can the `timeout` be passed in to the `HibernationManagerImpl` constructor... doing so could allow this to be a bit cleaner:

```suggestion
      KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
        hibernationManager.setEventTimeout(timeout);
      } else {
        a.setHibernationManager(kj::refcounted<HibernationManagerImpl>(
            a.getLoopback(), KJ_REQUIRE_NONNULL(a.getHibernationEventType(), timeout));
      }
```